commit 67aa7e6dd927c17103b3c5acb7eb50efb2372dab
Author: Martin K. Petersen <martin.petersen@oracle.com>
Date:   Sat Mar 5 17:52:02 2016 -0500

    sd: Fix discard granularity when LBPRZ=1
    
    commit 6540a65da90c09590897310e31993b1f6e28485a upstream.
    
    Commit 397737223c59 ("sd: Make discard granularity match logical block
    size when LBPRZ=1") accidentally set the granularity to one byte instead
    of one logical block on devices that provide deterministic zeroes after
    UNMAP.
    
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
    Reported-by: Mike Snitzer <snitzer@redhat.com>
    Reviewed-by: Ewan Milne <emilne@redhat.com>
    Reviewed-by: Bart Van Assche <bart.vanassche@sandisk.com>
    Fixes: 397737223c59e89dca7305feb6528caef8fbef84
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/scsi/sd.c b/drivers/scsi/sd.c
index bb669d3..cc84ea7 100644
--- a/drivers/scsi/sd.c
+++ b/drivers/scsi/sd.c
@@ -648,7 +648,7 @@ static void sd_config_discard(struct scsi_disk *sdkp, unsigned int mode)
 	 */
 	if (sdkp->lbprz) {
 		q->limits.discard_alignment = 0;
-		q->limits.discard_granularity = 1;
+		q->limits.discard_granularity = logical_block_size;
 	} else {
 		q->limits.discard_alignment = sdkp->unmap_alignment *
 			logical_block_size;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1280_67aa7e6dd927c17103b3c5acb7eb50efb2372dab.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1280_67aa7e6dd927c17103b3c5acb7eb50efb2372dab.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
